lbry/testcase.py

```diff
@@ -298,13 +298,15 @@ async def send_to_address_and_wait(self, address, amount, blocks_to_generate=0,
 
     async def generate_and_wait(self, blocks_to_generate, txids, ledger=None):
         if blocks_to_generate > 0:
-            watcher = (ledger or self.ledger).on_transaction.where(
+            watcher1 = (ledger or self.ledger).on_transaction.where(
                 lambda e: ((e.tx.id in txids and txids.remove(e.tx.id)), len(txids) <= 0)[-1]  # multi-statement lambda
             )
+            watcher2 = (ledger or self.ledger).on_header.where(self.blockchain.is_expected_block)
             self.conductor.spv_node.server.synchronized.clear()
             await self.blockchain.generate(blocks_to_generate)
```

Comment:
why not just use `self.generate()` here? there seems to be some code duplication happening...

Reply:
If you call IntegrationTestcase.generate(n) first, you are waiting for the height first, then the TX. I wasn't sure I should change the waiting order.

However, it works either way to stabilize the tests:
https://github.com/lbryio/lbry-sdk/pull/3667/commits/219b50650f0130285e05ecf83f589ad5dda0a6d7